lib/goth/server.ex

```diff
@@ -49,8 +32,25 @@ defmodule Goth.Server do
         Goth.HTTPClient.init({module, opts})
       end)
 
+    {async, opts} = Keyword.pop(opts, :async)
```

Comment:
the option should be called `:prefetch` and the possible values are `:sync` and `:async`. Please also update Goth.start_link/1 docs accordingly.